Only support stack 'o2cb' in init script o2cb. Stack 'user' support is not enabled in
WRL6 CGP, and we don't need to enable it in WRL7 too.

Source file /etc/init.d/functions rather than /lib/lsb/init-functions which
doesn't exist in wrlinux.  /etc/init.d/functions provides functions killproc
and pidofproc, and doesn't provides function start_daemon. It doesn't matter
that start_daemon is used by stack 'user' only.

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/vendor/common/o2cb.init.sh b/vendor/common/o2cb.init.sh
index 2e73108..926e61e 100755
--- a/vendor/common/o2cb.init.sh
+++ b/vendor/common/o2cb.init.sh
@@ -18,11 +18,11 @@
 # Force LC_ALL=C
 export LC_ALL=C
 
-# Let's try to use the LSB functions
-. /lib/lsb/init-functions
+# Let's try to use the init scripts functions
+. /etc/init.d/functions
 if [ $? != 0 ]
 then
-    echo "Unable to load LSB init functions" >&2
+    echo "Unable to load init scripts functions" >&2
     exit 1
 fi
 
@@ -72,6 +72,11 @@ VERSION=@@VERSION@@
 # Need this default
 [ -z "$O2CB_STACK" ] && O2CB_STACK=o2cb
 
+if [ "$O2CB_STACK" != "o2cb" ]; then
+    echo "Invalid cluster stack name \"$O2CB_STACK\""
+    exit 1
+fi
+
 configfs_path()
 {
     # Note that this is only valid *after* configfs is loaded
